staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -384,7 +400,7 @@ func generateResourcesAnnotation(resources sets.Set[schema.GroupVersionResource]
 }
 
 func (a ApplySet) FieldManager() string {
-	return fmt.Sprintf("%s-applyset-%s", a.toolingID.name, a.ID()) // TODO: validate this choice
+	return fmt.Sprintf("%s-applyset", a.toolingID.name)
```

Comment:
I'm thinking that the inclusion of the ID in the manager name is overly specific, particularly since this is only used to patch the parent itself (so it is a self-reference).

Reply:
Agreed that we need to think about it early.  I think the more valuable one to tie back to an applyset is the "child objects" - do we get to pick the field manager there (when using SSA)?  Or can we annotate in some way?  If we could explicitly that field manager back to say "these are the objects managed in this applyset, and these are the specific fields managed", then I think that would really be making SSA work for us...

Reply:
In theory we could use a separate field manager of our choice when setting our labels on the children instead of adding them to the object prior to the main operation. However, the field manager is set per-request: `PATCH https://127.0.0.1:51962/apis/company.com/v1/applysets/my-set?fieldManager=kubectl-applyset&fieldValidation=Strict&force=false` <-- note the first query param. So that would necessarily double the number of patches sent during an apply, which I suspect is unacceptable / not worth the benefit. 

We _could_ override the field manager name used for the main operation to be some value related to ApplySet, but that would have some undesired effects. First, there's CSA<->SSA migration/compatibility code (including on the server) that is triggered when the default names are used, so if we override the names, then ApplySet use makes a CSA->SSA upgrade non-rollback-safe. Second, the field manager name is exposed as a flag, so the user can override it.  

Another thing I realized in writing this out is that we _can't_  use the field manager here as in the main operation if we want to leave the door open to having the parent present in the main set. Allowing the parent to be in the main set would make ApplySet easier to use with custom resources, which may have required fields we can't guess values for. And if we used the same field manager name here, we'd effectively be requesting deletion of those extra fields.

Another option that occurred to me is using the field manager name from the main operation as the prefix, instead of the tooling name. That could work for SSA and for the case where the user overrides the name, but CSA has a different default name, and we don't want a CSA->SSA migration to have any effect on this (plus the CSA name literally has '-client-side-apply` in it, which is inaccurate for us).

Reply:
Good points all.  I was thinking about the "main" apply.   TBH my instinct was to have one field manager name for everything we apply, though I do now see that will introduce some very tricky edge cases for the parent.

I think CSA -> SSA means that we can't really mess with this, unless we were willing to make prunev2 SSA-only. 
 Maybe we should think about - easy enough to relax later and might avoid opening some as-yet-unknown cans of worms?